moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp

```diff
@@ -335,14 +337,32 @@ class MoveGroupInterface::MoveGroupInterfaceImpl
     num_planning_attempts_ = num_planning_attempts;
   }
 
-  void setMaxVelocityScalingFactor(double max_velocity_scaling_factor)
+  void setMaxVelocityScalingFactor(double factor)
   {
-    max_velocity_scaling_factor_ = max_velocity_scaling_factor;
+    setMaxScalingFactor(max_velocity_scaling_factor_, factor, "velocity_scaling_factor", 0.1);
   }
 
-  void setMaxAccelerationScalingFactor(double max_acceleration_scaling_factor)
+  void setMaxAccelerationScalingFactor(double factor)
   {
-    max_acceleration_scaling_factor_ = max_acceleration_scaling_factor;
+    setMaxScalingFactor(max_acceleration_scaling_factor_, factor, "acceleration_scaling_factor", 0.1);
+  }
+
+  void setMaxScalingFactor(double& variable, const double target, const char* name, double fallback)
+  {
+    if (target > 1.0)
+    {
+      ROS_WARN_NAMED("move_group_interface", "Limiting max_%s (%.2f) to 1.0.", name, target);
+      variable = 1.0;
+    }
+    else if (target <= 0.0)
+    {
+      node_handle_.param<double>(std::string("robot_description_planning/default_") + name, variable, fallback);
+      ROS_WARN_NAMED("move_group_interface", "max_%s <= 0.0! Setting to default: %.2f.", name, variable);
```

Comment:
I would expect this to be reasonable behavior for this API, so I would prefer this as a `ROS_DEBUG` instead.

Reply:
I don't agree. If the user specifies a negative scaling factor, she should get a warning about that.

Reply:
I agree that clipping the value to [0, 1] is reasonable behavior, but I think users should be alerted if they attempt to set it outside of that range.

Reply:
Unrelated: shouldn't this be `ROS_WARN_NAMED_STREAM`?

Reply:
Using stream syntax would be more verbose. But, I don't mind.

Reply:
> I don't agree. If the user specifies a negative scaling factor, she should get a warning about that.

If you insist, I can agree with the negative factor.
In this case, I would prefer to have a separate branch for `0.0` though.
Otherwise there is no warning-free API to reset the factors to their defaults (aside from the Constructor)!